lib/service-manager.js

```diff
@@ -83,34 +83,38 @@ prototype.initOrUpdateDb = function(meshApp, callback) {
     return env;
   };
 
-  Executor.findById('1', function(err, executor) {
+  this._meshApp.dataSources.db.autoupdate(function(err) {
     if (err) return callback(err);
-    if (!executor) {
-      return self._initDb(stopStaleProcesses);
-    }
-
-    executor.APIPort = self._listenPort;
-    executor.save(stopStaleProcesses);
-  });
 
-  function stopStaleProcesses(err) {
-    if (err) return callback(err);
-    Process.find(function(err, procs) {
+    Executor.findById('1', function(err, executor) {
       if (err) return callback(err);
-      async.each(procs, stopProcess, callback);
+      if (!executor) {
+        return self._initDb(stopStaleProcesses);
+      }
+
+      executor.APIPort = self._listenPort;
+      executor.save(stopStaleProcesses);
     });
-  }
 
-  function stopProcess(proc, callback) {
-    // Anything other than null, undefined, or '' is a reason.
-    if (proc.stopReason != null && proc.stopReason !== '') {
-      return callback();
+    function stopStaleProcesses(err) {
+      if (err) return callback(err);
+      Process.find(function(err, procs) {
+        if (err) return callback(err);
+        async.each(procs, stopProcess, callback);
+      });
     }
-    debug('mark stopped: pid %j wid %j', proc.pid, proc.workerId);
-    proc.stopReason = 'StrongLoop Process Manager was stopped';
-    proc.stopTime = new Date();
-    proc.save(callback);
-  }
+
+    function stopProcess(proc, callback) {
```

Comment:
Was it necessary to move this inside the callback? It looks like a whitespace-only change, since everything the function uses is passed in as arguments.